File: Ordenação/Python/bubbleSort.py
def bubble(vt): # Função do BubbleSort, de parametro recebe um vetor.

	for i in range(0,len(vt)): # Para i no alcance de 0 até o tamanho do vetor.

		for j in range(i+1, len(vt)): # Para j no alcance de i+1 (n precisa analisar ele mesmo, caso j=0) até o tamanho do vetor.

			if(vt[i]>vt[j]): # Se o valor do vet na posição anterior for maior que o vet sucessor, troca!

				# Trocas de Valores.
				aux = vt[i]
				vt[i] = vt[j]
				vt[j] = aux

vet = [] # Inicializa a Lista/Vetor.
n = int(input("Digite o Tamanho do Vetor: ")) # "Restringe" o Tamanho de elementos.

for t in range(0,n): # Para t no alcance de 0 até o número inserido pelo user em "n".
	val = int(input("Digite o {}º elemento: ".format(t+1))) # Pega do usuario o valor pra add no vetor
	vet.append(val) # Adiciona o valor digitado no vetor

print("Vetor antes do BubbleSort: {}".format(vet))

bubble(vet) # Faz o BubbleSort

print("Vetor depois do BubbleSort: {}".format(vet))

File: Ordenação/Cpp/bubbleSort.cpp
// Algoritmo para BubbleSort Crescente e Decrescente

#include <iostream>

using namespace std;

int main(){

	int n, aux, aux2;

	cout << "Digite o tamanho do vetor: ";
	cin >> n;

	int vet[n];

	for(int k=0; k<n; k++){
		cout << "Digite o " << k+1 << "º " << "elemento: ";
		cin >> vet[k];
	}

	cout << "Antes do BubbleSort: " << endl;

	for(int a=0; a<n; a++){
		cout << vet[a] << " ";
	}

	cout << endl;

	for(int i=0; i<n; i++){
		for(int j=i+1; j<n; j++){

			if(vet[i] > vet[j]){
				aux = vet[i];
				vet[i] = vet[j];
				vet[j] = aux;
			}

		}
	}

	cout << "Depois do BubbleSort Crescente: " << endl;

	for(int dc=0; dc<n; dc++){
		cout << vet[dc] << " ";
	}

	cout << endl;


	for(int l=0; l<n; l++){
		for(int c=l+1; c<n; c++){

			if(vet[l] < vet[c]){
				aux2 = vet[l];
				vet[l] = vet[c];
				vet[c] = aux2;
			}

		}
	}

	cout << "Depois do BubbleSort Decrescente: " << endl;

	for(int dd=0; dd<n; dd++){
		cout << vet[dd] << " ";
	}

	cout << endl;

	return 0;
}
